src/main/java/com/microsoft/sqlserver/jdbc/TimeoutPoller.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Microsoft JDBC Driver for SQL Server Copyright(c) Microsoft Corporation All rights reserved. This program is made
+ * available under the terms of the MIT License. See the LICENSE file in the project root for more information.
+ */
+
+package com.microsoft.sqlserver.jdbc;
+
+import java.util.ArrayList;
+import java.util.Iterator;
+import java.util.List;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * Thread that runs in the background while the mssql driver is used that can timeout TDSCommands Checks all registered
+ * commands every second to see if they can be interrupted
+ */
+final class TimeoutPoller implements Runnable {
+    private List<TimeoutCommand<TDSCommand>> timeoutCommands = new ArrayList<>();
+    final static Logger logger = Logger.getLogger("com.microsoft.sqlserver.jdbc.TimeoutPoller");
+    private static volatile TimeoutPoller timeoutPoller = null;
+
+    static TimeoutPoller getTimeoutPoller() {
+        if (timeoutPoller == null) {
+            synchronized (TimeoutPoller.class) {
+                if (timeoutPoller == null) {
+                    // initialize the timeout poller thread once
+                    timeoutPoller = new TimeoutPoller();
+                    // start the timeout polling thread
+                    new Thread(timeoutPoller, "mssql-jdbc-TimeoutPoller").start();

```

Comment:
Can we change this to something along the lines of the following:
```java
Thread t = new Thread(timeoutPoller, "mssql-jdbc-TimeoutPoller");
t.setDaemon(true);
t.start();
```
I was fiddling around with the new changes, and in certain scenarios, this infinitely running thread is able to block java's DestroyVM thread. That thread performs a join() on this thread, and since this thread never terminates, it'll just hang the jvm. I'm not quite sure why it doesn't happen in any of the JUnit tests.

Reply:
It doesn't happen in the junit tests because I believe the junit runner (well at least RemoteTestRunner) calls System.exit(), which terminates the jvm regardless of daemon/non-daemon threads (when main method finishes and you still have non daemon threads, you will get into issues where the jvm process does not exit).

Regardless this seems like the right fix to make. The only possible reason to not make this is a daemon thread is if we want to build logic into the poller such that on jvm exit, we timeout any queries that are registered, then exit the thread.

If this behavior is desired, please let me know. Otherwise I will make this a daemon thread.

Reply:
done